server.go

```diff
@@ -602,13 +752,18 @@ func (s *Server) Stop() {
 	s.lis = nil
 	cs := s.conns
 	s.conns = nil
+	for c := range s.hconns {
+		go c.Close()
```

Comment:
+1 to a proper `Stop()` and not relying on a crash/exit. But maybe that should be a separate PR.